src/app/libp2p_helper/src/libp2p_helper/main.go

```diff
@@ -821,6 +834,10 @@ func (cs *sendStreamMsgMsg) run(app *app) (interface{}, error) {
 	app.StreamsMutex.Lock()
 	defer app.StreamsMutex.Unlock()
 	if stream, ok := app.Streams[cs.StreamIdx]; ok {
+		msgLen := uint64(len(data))
+		lenBytes := uint64ToLEB128(msgLen)
+		data = append(lenBytes, data...)
+
 		n, err := stream.Write(data)
```

Comment:
Don't we need to manually write an EOF here if we do not close the stream?

Reply:
no, the reader will automatically get an EOF if there's no more data left to read on the stream

Reply:
So for clarity of the workflow here: we would only receive a single EOF after the stream is closed? (which makes sense, I just got confused)

Reply:
no, we would receive an EOF when there is no more data left to read (not the same as the stream closing). if the remote side closes the stream we will get `err == "stream reset`

Reply:
I thought with the way go's stream writers work (from reading some docs), it only sends EOF when you close the writer, and does not send EOF between each subsequent write. Is this not the case? Or should we be manually sending an EOF between each message? The docs around the EOF semantics are a bit vague, so I'm a little confused.